src/main/flight/pid.c

```diff
@@ -466,7 +466,7 @@ void pidInitFilters(const pidProfile_t *pidProfile)
     pt1FilterInit(&antiGravityThrottleLpf, pt1FilterGain(ANTI_GRAVITY_THROTTLE_FILTER_CUTOFF, dT));
 
     ffBoostFactor = (float)pidProfile->ff_boost / 10.0f;
-    ffJerkLimitInverse = pidProfile->ff_jerk_limit ? 1.0f / ((float)pidProfile->ff_jerk_limit / 10.0f) : 0.0f;
+    ffSpikeLimitInverse = pidProfile->ff_spike_limit ? 1.0f / ((float)pidProfile->ff_spike_limit / 10.0f) : 0.0f;
```

Comment:
The math here seems a little off - particularly for `ff_spike_limit` values below 10. Is this really intended?

![ffSpikeLimitInverse](https://user-images.githubusercontent.com/17088539/64480775-f0b46100-d19b-11e9-9385-03c9fbad301c.png)

Reply:
the intended attenuation curve from boost to clip, with the default threshold of 40, should be like this:
![Screen Shot 2019-09-08 at 10 44 04](https://user-images.githubusercontent.com/11737748/64481725-b2fc1a80-d225-11e9-8d6c-dfdb6503a1dd.jpg)

Reply:
Alright, there is a mistake here.  When it was refactored, the value for spike limit was assumed by Joe to be 10 times larger than needed, like some other FF values, and hence that it should be divided by 10.  The resulting clip factors in the tests above were therefore from threshold values effectively of 4, not of 40, resulting in 10x stronger suppression of the boost factor than intended.  The grey dots here show the suppression applied when the threshold is effectively 4, compared to 40.

![Screen Shot 2019-09-08 at 10 44 04](https://user-images.githubusercontent.com/11737748/64481871-8d711000-d229-11e9-90f4-46cae36dd456.jpg)

Reply:
As it turns out, this was a good thing.  All the testing was done with this error in place, and it seems to have worked out better like this.
Most useful input boost values are very small - less than 5 in absolute terms.  For instance, a single RC step change in setpoint from 90 to 180 results in a boost amount of 2.3.  Early in a move, boost amounts are typically 1 or less.  In contrast, going from 170 to 420 deg/s in one RC step is a boost of about 12.  This is the kind of boost/step amount that we get as we approach the end of the stick movement when starting a flip, and we want to attenuate that.  Boost spikes of about 40 are typical from RC steps.
So as it turns out, the grey curve showing the result of our current code - with the /10 - gives more appropriate attenuation than the early versions I was testing.
This explains why the re-factored version suppressed spikes better than I had anticipated.
Here is a comparison of the current code on the right vs code without the /10 on the left (effective threshold of 400 on left, 40 on right).  
![Screen Shot 2019-09-08 at 11 37 14](https://user-images.githubusercontent.com/11737748/64482235-2d7e6780-d231-11e9-901a-ac6697fa448a.jpg)

Reply:
I think we should leave the code as is.  
The error was serendipitous.
40 is a good safe value for the threshold with the code is.  
It gives strong spike suppression and reasonable transmission of desirable boost elements.
Threshold at 40 probably allows usage with the averaging off without major issues on quads with good RC signals, and values up to 100 or 200 with averaging on would be OK I think.

Reply:
You're missing the point of the concern. The issue is that the parameter `ff_spike_limit` is allowed to range from 0 (off) through 255. The math I pointed out has a problem if `ff_spike_limit` is set to values less than 10. In those cases `ffSpikeLimitInverse` becomes a multiplier ( > 1) instead of a reduction. The graph I posted shows the resulting `ffSpikeLimitInverse` values based on possible user settings.

Reply:
Oh - TWO silly errors when refactored!  
Indeed.

Reply:
This shows similar inputs with `ff_spike_limit` at 40, on left, and 4, on right.  When set to 4 there is effectively no boost. The resulting FF trace at the top is, effectively, normal feed forward with no boost.
Boost of 25 here is higher than default of 15; the effect is stronger than normal.
Note how with the default limit of 40, boost leads to:
- much stronger earlier feed forward
- somewhat more feed forward effect overall
- feed forward reversal (negative feed forward) when sticks stop
- significantly greater spikes with RC steps
These stops are all hard, with rounded stops, boost drops the feed forward smoothly and negatively earlier on.
![Screen Shot 2019-09-08 at 12 52 26](https://user-images.githubusercontent.com/11737748/64482596-18590700-d238-11e9-9565-8394336b449f.jpg)

Reply:
This image shows what happens when the 2-point averaging is turned off.
The whole feed forward response becomes significantly more aggressive, with spikes that are about twice as big with RC steps.  The boost effect comes on earlier and goes away quicker, leading to less overshoot and greater setpoint precision in flight.  Right side is averaging off.
![Average on vs off](https://user-images.githubusercontent.com/11737748/64482691-31fb4e00-d23a-11e9-9135-6d7f44464c09.jpg)
However the overall increase in noise with typical RC signals means that turning averaging off should only be done with very clean RC data.  Right side is averaging off.
![Nouse Avg on vs off](https://user-images.githubusercontent.com/11737748/64482702-58b98480-d23a-11e9-91ff-75581499aba5.jpg)

Reply:
These are some real flips using `ff_boost = 15` and `ff_spike_limit = 50`, with averaging active on the left but not on the right.  RC signal is solid.  Both outcomes are excellent, can't see any meaningful difference though:
![Flips_AvgOnvsAvgOff](https://user-images.githubusercontent.com/11737748/64482813-09745380-d23c-11e9-8018-82c811889ed6.jpg)
The difference can be seen here with very rapid reversals - but only if we look very closely.  Without the feed forward extending effect of the averaging, the feed forward trace reverses harder when the sticks reverse, leading to less overshoot and harder overshoot control.  The motor traces are more aggressive with averaging off.
![Reversals_AVGvsON](https://user-images.githubusercontent.com/11737748/64482821-3aed1f00-d23c-11e9-96e4-fa3fe2e16cec.jpg)

Reply:
Ok, but I still think the results will be a bit unexpected if users set `ff_spike_limit` < 10 as the change in the calculations become dramatically different.